addon/authenticators/osf-cookie.js

```diff
@@ -0,0 +1,54 @@
+import Base from 'ember-simple-auth/authenticators/base';
+import config from 'ember-get-config';
+
+import { ajaxAuth } from 'ember-osf/utils/ajax-helpers';
+
+/**
+ * @module ember-osf
+ * @submodule authenticators
+ */
+
+/**
+ * Ember-simple-auth compatible authenticator based on session cookie.
+ *
+ * Intended to be used with the authorizer of the same name.
+ *
+ * @class OsfCookieAuthenticator
+ * @extends ember-simple-auth/BaseAuthenticator
+ */
+export default Base.extend({
+    _test() {
+        return ajaxAuth({
+            method: 'GET',
+            url: `${config.OSF.apiUrl}/${config.OSF.apiNamespace}/users/me/`,
+            dataType: 'json',
+            contentType: 'application/json',
+            xhrFields: {
+                withCredentials: true
+            }
+        }).then(function(res) {
+            return res.data;
+        });
+    },
+    restore(/* data */) {
+        return this._test().fail(this.invalidate);
+    },
+
+    /**
+     * Send a request to the flask application to trigger invalidation of session remotely
+     * @method invalidate
+     */
+    invalidate() {
+        // Can't do this via AJAX request because it redirects to CAS, and AJAX + redirect = CORS issue
+        window.location = `${config.OSF.url}logout/`;
```

Comment:
In principle, the OSF keeps track of its own sessions. Hitting CAS alone would not guarantee that the session cookie was removed/invalidated. Here's the method the OSF uses to logout- it hits CAS, but also does other things.
https://github.com/abought/osf.io/blob/a9054b02310f31223933a9f62116574176fce1d5/framework/auth/views.py#L277-L277

...Come to think of it, I'm not sure whether/how that causes the cookie to be actually invalidated server side (if someone presents it as a bearer instrument later). But it does at least remove the cookie from the user's computer.